Player movement

Eryngii01 opened the pull request:
Implemented 4 player states (WalkingState, RunningState, HidingState, DyingState) to handle the player movement behaviour on Update with both OnEnter and OnExit methods for future extension in terms of adding animations, audio, VFX, etc. 

Also implemented a HRGauge class that is regulated by the PlayerController class to handle all operations regarding increasing and decreasing the heart rate gauge as well as the threshold, which includes the handling of the player death state change.

Eryngii01 committed:
Set up InputActions and simple architecture

Added InputActions asset for player input and created a simple PlayerState architecture

Eryngii01 committed:
Merge branch 'main' into player-movement

Eryngii01 committed:
Implemented movement per states and player death

Implemented the different types of movement behaviour per the different states for the player and added the player death state according to the Heart Rate Gauge

Eryngii01 committed:
Tested state behaviour and added documentation

Made minor adjustments to ensure the state behaviour works and added some documentation TODOs for future expansions

Eryngii01 committed:
Removed Debug.Log calls

adashivers commented on Assets/Code/Scripts/Player/WalkingState.cs:
```diff
@@ -0,0 +1,37 @@
+        
+        public void Movement(Transform player, HRGauge heartRate, Action playerDeath, float baseSpeed)
+        {
+            float speed = baseSpeed + _speedMultiplier;
```
did you mean to add or multiply here?

Eryngii01 replied:
Ah I meant to add here but made the variable name a little misleading so I'll adjust that

eon704 commented on .idea/.idea.to-jam-23/.idea/.gitignore:
It looks like this .gitignore should be in the root directory 😅

eon704 replied:
Also I believe this folder should ignored as well

eon704 reviewed (approved):
.idea folder will need to be removed, otherwise LGTM!

Eryngii01 committed:
Adjusted attribute names and gitignore

Adjusted the attribute names according to the PR comments and fixed the gitignore for the PR

eon704 committed:
Updated .gitignore
